src/main/java/org/jadice/filetype/matchers/PDFMatcher.java

```diff
@@ -74,6 +75,7 @@ public class PDFMatcher extends Matcher {
   public boolean matches(final Context context) {
     SeekableInputStream sis = context.getStream();
     try {
+      long fileLength = getFileLength(sis);

```

Comment:
Sieht an sich gut aus. Aber möchte ich wirklich jedes mal (also ohne Config-Flag) und meinem Verständnis nach nicht nur bei PDF, den gesamten Stream lesen um die Länge zu bekommen?

Reply:
Der PDFMatcher wird nur ausgeführt, wenn das Dokument mit `PDF` oder `%PDF` beginnt. Siehe `PDFMatcher` in https://github.com/levigo/filetype-analyzer/blob/master/src/main/resources/magic.xml
Vielleicht könnte man trotzdem vorher prüfen, ob es eine Signatur enthält.

Reply:
Ich hatte das absichtlich so früh im Code gemacht, weil ich mir bei Folgendem nicht sicher war:  
Wenn man übprüft, ob das Dokument eine Signatur enhält, ist man ja schon in dem Teil, wo das Dokument per `PDDocument.load(sis)` geladen wurde. Da bin ich mir nicht sicher, ob man den `SeekableInputStream` in dem Bereich, wo man das `PDDocument` Objekt verwendet, anrühren (insbesondere die Position darin verändern) darf.  
Weiß das jemand von euch?  
Im Prinzip gehts mir darum, ob `PDDocument.load(sis)` eager oder lazy funktioniert, also ob es direkt alle nötigen Informationen in den Speicher lädt, oder bei manchen Anfragen auch nochmal auf den Stream zugreift.

Reply:
Im Javadic von `PDDocument.load(sis)` heißt es

> Parses a PDF. The given input stream is copied to the memory to enable random access to the pdf. Unrestricted main memory will be used for buffering PDF streams.

Daher gehe ich davon aus, dass PDFBox nicht mehr auf den Stream zugreift. Folglich könntest du die Länge auch erst dann bestimmen, wenn klar ist, dass das Dokument eine Signatur hat.

Reply:
Danke, hab ich so umgesetzt